lib/engines/hcktest/hcktest.rb

```diff
@@ -143,29 +143,30 @@ def run_clients(scope, run_opts = {})
                                 this platform is incorrect'
     end
 
-    def configure_clients
+    def configure_and_synchronize_clients
       run_only = @project.options.test.manual && @project.options.test.driver_path.nil?
 
       @clients.each_value do |client|
         client.configure(run_only:)
       end
+
+      @clients.each_value(&:synchronize)
     end
 
     def configure_setup_and_synchronize
+      return configure_and_synchronize_clients unless @studio.tools.nil?
+
       @studio.configure(@platform['clients'])
-      configure_clients
-      @clients.each_value(&:synchronize)
+      configure_and_synchronize_clients
       @studio.keep_snapshot
       @clients.each_value(&:keep_snapshot)
     end
 
-    def run_and_configure_setup(scope)
+    def run_clients_and_configure_setup(scope, client_run_opts = {})
       retries = 0
       begin
         scope.transaction do |tmp_scope|
-          run_studio tmp_scope
-          sleep 5 until @studio.up?
-          run_clients tmp_scope, keep_alive: true
+          run_clients tmp_scope, keep_alive: true, **client_run_opts
```

Comment:
I see a problem with the current implementation of HCKClient recreation. When we create a new HCKClient it will call setup_manager to create a snapshot again. In this case, we lose information from the previous run (can capture files from a real VM if needed). 

There are 2 ways:
- add a new option to reuse existing snapshot
- add a new option to rename snapshots according to the test group

@akihikodaki What do you think?

Reply:
I'm for the former. We don't have a reason to create a snapshot for each test group while we don't create one for each test.